thundra/wrappers/django/django_wrapper.py

```diff
@@ -0,0 +1,126 @@
+import logging
+import traceback
+import uuid
+from functools import wraps
+
+from thundra import constants, configure
+from thundra.application.global_application_info_provider import GlobalApplicationInfoProvider
+from thundra.config import config_names
+from thundra.config.config_provider import ConfigProvider
+from thundra.context.execution_context_manager import ExecutionContextManager
+from thundra.context.plugin_context import PluginContext
+from thundra.context.tracing_execution_context_provider import TracingExecutionContextProvider
+from thundra.plugins.invocation import invocation_support
+from thundra.wrappers import wrapper_utils
+from thundra.wrappers.base_wrapper import BaseWrapper
+from thundra.wrappers.django import django_executor
+
+logger = logging.getLogger(__name__)
+
+
+class DjangoWrapper(BaseWrapper):
+
+    def __init__(self, api_key=None, disable_trace=False, disable_metric=True, disable_log=True, opts=None):
+        try:
+            from django.conf import settings
+            django_settings = getattr(settings, 'THUNDRA', {})
+            configure({'config': django_settings})
+
+            api_key = None
+            for var in django_settings:
+                if var.lower() == 'thundra.apikey' and not api_key:
+                    api_key = django_settings.get(var)
+        except:
+            pass
+
+        super().__init__(api_key, disable_trace, disable_metric, disable_log, opts)
+        self.application_info_provider = GlobalApplicationInfoProvider()
+        ExecutionContextManager.set_provider(TracingExecutionContextProvider())
+        self.plugin_context = PluginContext(application_info=self.application_info_provider.get_application_info(),
+                                            request_count=0,
+                                            executor=django_executor,
+                                            api_key=self.api_key)
+
+        self.plugins = wrapper_utils.initialize_plugins(self.plugin_context, disable_trace, disable_metric, disable_log,
+                                                        config=self.config)
+
+    def before_request(self, request):
+        # Set application info
+        self.application_info_provider.update({
```

Comment:
Do we need to update before every request?

Reply:
Additionally, this code block seems not specific to Django (exception app class name). Can we move it to a common place?